ds/ds.php

```diff
@@ -1272,21 +1326,126 @@ public function copy(): Collection
         {
         }
 
+        /**
+         * Returns the result of removing all keys from the current instance that are present in a given map.
+         *
+         * A \ B = {x ∈ A | x ∉ B}
+         *
+         * @param Map $map The map containing the keys to exclude in the resulting map.
+         *
+         * @return Map The result of removing all keys from the current instance that are present in a given map.
+         *
+         * @link https://www.php.net/manual/en/ds-map.diff.php
+         */
+        public function diff(Map $map): Map
+        {
+        }
+
+        /**
+         * Creates a new map using a callable to determine which pairs to include
+         *
+         * @param callable $callback Optional callable which returns TRUE if the pair should be included, FALSE
```

Comment:
please replace Optional to [optional] to make param interpreted as optional

Reply:
This is not intended to mark the callable as optional. What I forgot is to add the default value to an optional callable (null). The line you marked is the actual text of the documentation. https://www.php.net/manual/en/ds-map.filter.php